### ISSUE TITLE
genfromtxt fails in latest 1.19.0 with some parameters

### ISSUE THREAD

**Issue by FedeMiorelli** [numpy/numpy#16728]
Since updating to numpy 1.19.0, the function genfromtxt fails in some cases, parsing floating point numbers as complex values (with 0 imaginary part) and failing trying to convert them later to float.

This occurs only with some combination of parameters. In this example, it fails if the user specifies the field names but succeeds otherwise.

Below is a short code to reproduce and a small data file to test.
**[test.txt](https://github.com/numpy/numpy/files/4862568/test.txt)**

### Reproducing code example:

```python
import numpy as np
filename = 'test.txt' # link to the file is above
data1 = np.genfromtxt(filename, usecols=(3,))  # This works
data2 = np.genfromtxt(filename, usecols=(3,), names=['a'])  # This doesn't

```

### Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\WinPython-64bit-3.8.3.0\python-3.8.3.amd64\lib\site-packages\numpy\lib\npyio.py", line 2210, in genfromtxt
    output = np.array(data, dtype)
TypeError: can't convert complex to float
```

### Output with numpy 1.17.2
```
>>> np.genfromtxt('test.txt', usecols=(3,))
array([2.73, 2.73, 2.59, 2.59, 2.65, 2.65])
>>> np.genfromtxt('test.txt', usecols=(3,), names=['a'])
array([(2.73,), (2.73,), (2.59,), (2.59,), (2.65,), (2.65,)],
      dtype=[('a', '<f8')])
```

### Output with numpy 1.19.0
```
>>> np.genfromtxt('test.txt', usecols=(3,))
array([2.73, 2.73, 2.59, 2.59, 2.65, 2.65])
>>> np.genfromtxt('test.txt', usecols=(3,), names=['a'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\WinPython-64bit-3.8.3.0\python-3.8.3.amd64\lib\site-packages\numpy\lib\npyio.py", line 2210, in genfromtxt
    output = np.array(data, dtype)
TypeError: can't convert complex to float
>>>                                                  
```

### Numpy/Python version information:

`1.19.0 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)]`

**Comment by eric-wieser** [comment]
@seberg, this looks likely to be related to the iotools converter stuff you and I looked at recently.

**Comment by charris** [comment]
Be nice to get this fixed for 1.19.1

**Comment by charris** [comment]
@seberg Ping.

**Comment by seberg** [comment]
Heh, just spend quite a bit of time trying to figure out why its not fixed yet, until I realized it is... Just forgot to mark gh-16242 for backport.

**Comment by seberg** [comment]
Wait, you actually backported that PR already yesterday, so it is already fixed.  Of course its a bit crazy, it goes through string to end up with floats, but that is how it always has been.

**Comment by charris** [comment]
> Wait, you actually backported that PR already yesterday,

I was looking at closed issues marked for 1.19.1 and found one that hadn't been backported.
